mesh/mesh.go

```diff
@@ -241,21 +257,34 @@ func (m *Mesh) ExtractUniqueOrderedTransactions(l *types.Layer) []*Transaction {
 	return MergeDoubles(txs)
 }
 
-func (m *Mesh) PushTransactions(oldBase types.LayerID, newBase types.LayerID) {
+func (m *Mesh) PushTransactions(oldBase, newBase types.LayerID) {
 	for i := oldBase; i < newBase; i++ {
-
 		l, err := m.GetLayer(i)
 		if err != nil || l == nil {
 			m.Error("could not get layer %v  !!!!!!!!!!!!!!!! %v ", i, err) //todo handle error
 			return
 		}
 
-		merged := m.ExtractUniqueOrderedTransactions(l)
-		x, err := m.ApplyTransactions(types.LayerID(i), merged)
+		validBlockTxs, invalidBlockTxs := m.ExtractUniqueOrderedTransactions(l)
+		numAppliedTxs, err := m.ApplyTransactions(i, validBlockTxs)
 		if err != nil {
-			m.Log.Error("cannot apply transactions %v", err)
+			m.With().Error("cannot apply transactions", log.Err(err))
 		}
-		m.Log.Info("applied %v transactions in new pbase is %d apply result was %d", len(merged), newBase, x)
+		if err := m.removeFromMeshTxs(validBlockTxs, invalidBlockTxs, i); err != nil {
```

Comment:
I'm not sure what the behaviour should be. Currently it's not great, since it stops removing on the first error, returns it and then continues with `PushTransactions`...

We should do one of these two options, IMO:
- Stop everything on failure and rollback (hard to implement)
- Continue trying to remove and continue the flow, logging any errors along the way (easy to implement and seems rather harmless to me)

WDYT?